src/Error/JsonApiExceptionRenderer.php

```diff
@@ -97,6 +97,10 @@ public function validation($exception)
         // set data and send response
         $this->controller->response->type('jsonapi');
         $this->controller->response->body($json);
+        $this->controller->response->cors($this->controller->request)
```

Comment:
I don't think exception render is the correct place to set CORS headers. That should be done for your request URL in middleware or beforeFilter ().